### PR TITLE
feat: Allow to include translations for extended profile fields

### PR DESCRIPTION
**Pull request by johanseto**
## Description

Pr migration.
(cherry picked from commit 5d03359b2b503ab1e15f67f90eb9df6ae47d4ccc)

## Testing instructions
Install eox-nelp and some other packages need it like. 
``` bash
pip install edx-event-routing-backends
pip install git+https://github.com/eduNEXT/eox-nelp.git
```
Add the middleware of eox-nelp.
```python
MIDDLEWARE += [
    "eox_nelp.middleware.ExtendedProfileFieldsMiddleware",
]
```
Abd in tenant config the following.

``` json
    "REGISTRATION_EXTRA_FIELDS": {
        "city": "hidden",
        "confirm_email": "hidden",
        "country": "optional",
        "first_name": "required",
        "gender": "optional",
        "goals": "optional",
        "hobby": "optional",
        "honor_code": "required",
        "last_name": "required",
        "level_of_education": "optional",
        "mailing_address": "hidden",
        "movie": "optional",
        "sport": "optional",
        "terms_of_service": "hidden",
        "year_of_birth": "optional"
    },
    "extended_profile_fields": [
        "hobby",
        "sport",
        "movie"
    ],
    "extended_profile_fields_translations": {
        "ar": {
            "hobby": "\u0647\u0648\u0627\u064a\u0629",
            "movie": "\u0641\u064a\u0644\u0645",
            "sport": "\u0631\u064a\u0627\u0636\u0629"
        }
    }
```
Check translation works in account settings view changing the language.
## Extra information

This need eox-nelp, to use the middleware of extended profile fields. For account settings view, the middleware is working but the register modal is not showing the custom fields using the `extended_profile_fields`.

Screenshot from 2024-01-23 17-01-36

### CODE REVIEW

File: openedx/core/djangoapps/user_api/accounts/settings_views.py [hunk 1]
```diff
@@ -270,6 +271,13 @@ def _get_extended_profile_fields():
         "profession": _("Profession"),
         "specialty": _("Specialty")
     }
+    request = theming_helpers.get_current_request()
+    extended_profile_fields_translations = configuration_helpers.get_value(
+        'extended_profile_fields_translations',
+        {},
+    )
+    translations = extended_profile_fields_translations.get(request.LANGUAGE_CODE, {})
+    field_labels_map.update(translations)
```
Comment (andrey-canon) [inline openedx/core/djangoapps/user_api/accounts/settings_views.py:280]:
```suggestion
    request = theming_helpers.get_current_request()
    
    if request:
        extended_profile_fields_translations = configuration_helpers.get_value(
            'extended_profile_fields_translations',
            {},
        )
        translations = extended_profile_fields_translations.get(request.LANGUAGE_CODE, {})
        field_labels_map.update(translations)
```

I thin this change fixes tests

**Comment by johanseto** [comment]
Tested after test fix and still working.
image

**Review by andrey-canon — APPROVED** [review]
The middleware is not required since this is specific for account settings, anyway worked as expected 

image

**Comment by johanseto** [comment]
Working in palm alpha.
Peek 2024-02-22 18-41

**Pull request merged**